java/arcs/android/storage/database/AndroidSqliteDatabaseManager.kt

```diff
@@ -50,9 +53,11 @@ class AndroidSqliteDatabaseManager(context: Context) : DatabaseManager {
             .forEach { it.reset() }
     }
 
-    override suspend fun removeExpiredEntities() {
-        registry.fetchAll()
-            .map { getDatabase(it.name, it.isPersistent) as DatabaseImpl }
-            .forEach { it.removeExpiredEntities() }
+    override suspend fun removeExpiredEntities(): Job = coroutineScope {
```

Comment:
what difference does it make for the caller?
do you need to .join() in PeriodicCleanupTask?